add `has-preprocess-tab` attribute to fetch-button

ncittm opened the pull request:
By default show only the text `Go to the "Run Analysis" tab to continue.`,
"Preprocess" is only added to this text if `has-preprocess-tab` is true.

ncittm committed:
add `has-preprocess-tab` attribute to fetch-button

By default show only the text `Go to the "Run Analysis" tab to continue.`,
"Preprocess" is only added to this text if `has-preprocess-tab` is true.

sherzinger commented on test/unit/javascript/fetchButtonDirectiveTests.js:
```diff
@@ -96,6 +96,19 @@ describe('fetchButton', function() {
+
+    it('should show `Task complete! Go to the "Preprocess" or "Run Analysis" tab to continue.`', function() {
+        // NOTE overriding parts of `beforeEach` here
+        var html = '<fetch-button concept-map="conceptMap" allowed-cohorts="allowedCohorts" has-preprocess-tab="true"></fetch-button>';
```
You don't need to overwrite beforeEach. Just expand the model with a variable "hasPreprocessTab", link fetch-button to it, default it to false in _prepareScope and set it to true if the respective argument is given otherwise.

ncittm replied:
Hi Sascha,

zwei Fragen:

1.
wenn ich das richtig sehe werden $rootScope.conceptMap und
$rootScope.hasPreprocessTab gar nicht benutzt, denn in _prepareScope
wird der entsprechende Wert von .isolateScope() gesetzt!? Oder sind
isolateScope und $rootScope das gleiche (die scopes immer noch ein
bisschen schleierhaft)?

2.
fuer _prepareScope muss ich einen default setzen, diese Funktionalitaet
in javascript scheint noch zu neu zu sein:
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Default_parameters

Was ist Dir lieber:
1. den call zu _prepareScope ueberall anpassen
2. In _prepareScope scope.hasPreprocessTab immer auf das uebergebene
Argument setzen, dann wird es `undefined` wenn ein Funktionsaufruf
diesen Wert nicht uebergibt (sollte auch klappen)
3. in _prepareScope `if (hasPreprocessTab != undefined)... `

Du bist der Meister von master, Du entscheidest. ;-)

Cheers
  Nils

On 06/09/16 10:37, Sascha Herzinger wrote:
> In test/unit/javascript/fetchButtonDirectiveTests.js
> <https://github.com/transmart/SmartR/pull/49#discussion_r77593422>:
> 
>> @@ -96,6 +96,19 @@ describe('fetchButton', function() {
>>      it('should have the correct scope when successful with showSummaryStats', function() {
>>          _prepareScope(1, [1,2], true, {foo: {concepts: ['concept'], valid: true}});
>>          _clickButton('resolve()', 'resolve({result: {allSamples: 1337, subsets: null}})');
>> +        expect(element.find('span').text()).toBe('Task complete! Go to the "Run Analysis" tab to continue.');
>> +        expect(element.isolateScope().running).toBe(false);
>> +        expect(element.isolateScope().loaded).toBe(true);
>> +        expect(element.isolateScope().allSamples).toEqual(1337);
>> +    });
>> +
>> +    it('should show `Task complete! Go to the "Preprocess" or "Run Analysis" tab to continue.`', function() {
>> +        // NOTE overriding parts of `beforeEach` here
>> +        var html = '<fetch-button concept-map="conceptMap" allowed-cohorts="allowedCohorts" has-preprocess-tab="true"></fetch-button>';
> 
> You don't need to overwrite beforeEach. Just expand the model with a
> variable "hasPreprocessTab", link fetch-button to it, default it to
> false in _prepareScope and set it to true if the respective argument is
> given otherwise.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/transmart/SmartR/pull/49/files/d303a9622b8b56c2f2f1a5e06fe47bb0f2f20966#r77593422>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARNlbNyckiBb9IZUXd48UBXwHm_0zJ8uks5qnSZEgaJpZM4J1BDn>.
> 


-- 
Dr. Nils CHRISTIAN
Developer & Bioinformatician

ITTM S.A.
(Information Technology for Translational Medicine)
9 avenue des Hauts-Fourneaux
L-4362 Esch-sur-Alzette
Luxembourg
http://www.ittm-solutions.com

-----
The information transmitted is intended only for the person or entity to
which it is addressed and may contain confidential and/or privileged
material. Any review, retransmission, dissemination or other use of, or
taking of any action in reliance upon, this information by persons or
entities other than the intended recipient is prohibited. If you
received this in error, please contact the sender and delete the
material from any computer.
-----

sherzinger replied:
1. Der einzige Grund, warum es im rootScope gesetzt werden muss, ist dass diese beiden variablen nicht optional sind. D.h. wenn du sie beim Erstellen des elements nicht gesetzt hast, kracht es. Da hasPreprocessTab optional ist, kannst du den $rootScope ignorieren und musst auch den HTML snippet nicht veraendern. Hab eben am Telefon nicht daran gedacht.

2. Du erweiterst die Parameter um hasPreprocessTab und checkst ob es definiert ist, ansonsten setzt du es auf false. Was du in 3. geschrieben hast geht in die richtige Richtung, aber die Condition macht mir Gaensehaut. Als Daumenregel: Benutze kein != oder == in JS. Gib mal null == undefined in die Console ein. 
Mach lieber typeof hasPreprocessTab === 'undefined'

ncittm committed:
`has-preprocess-tab`s value is set to variable in $rootScope, which is modifed in _prepareScope

Instead of overriding parts of `beforeEach`, `has-preprocess-tab` is bound
to a variable in $rootScope. Its value is modified in _prepareScope.
